src/feedcontainer.cpp

```diff
@@ -9,95 +9,120 @@
 
 namespace newsboat {
 
+class UnreadArticeCountComparator {
+	FeedSortStrategy sort_strategy;
+
+public:
+	UnreadArticeCountComparator(const FeedSortStrategy& sort_strategy)
+		: sort_strategy(sort_strategy) {}
+
+	bool operator()(const std::shared_ptr<RssFeed>& a,
+		const std::shared_ptr<RssFeed>& b)
+	{
+		int diff = a->unread_item_count() - b->unread_item_count();
+		bool result = false;
+		if (diff != 0) {
+			if (sort_strategy.sd == SortDirection::DESC) {
+				diff *= -1;
+			}
+			result = diff > 0;
+		}
+		return result;
```

Comment:
Can this code be moved into `sort_feeds`? It's a bit weird to have a separate class for this when we can have it inline (like we do for all other sorts).

Also, I think `diff` is superfluous. Why not initialize `result` directly, like in other sorts?

Reply:
because result depends on three possible states of diff, not two (like boolean). maybe it cant be done better :thinking:

Reply:
I don't see a reason to special-case `diff == 0`. The line `result = diff > 0` already handles this case correctly, returning `false`; the change of sort strategy doesn't change the result, because `0 * -1 == 0`. So I'm pretty sure you can drop the `if` and initialize the `result` right away.